jwt/jwt.go

```diff
@@ -118,6 +118,44 @@ func getJWK(ctx context.Context, kid string) (*clerk.JSONWebKey, error) {
 	return nil, fmt.Errorf("no jwk key found for kid %s", kid)
 }
 
+// Returns the JSON web key set. Tries a cached value first, but if
+// there's no value or the entry has expired, it will fetch the set
+// from the API and cache the value.
+func getJWKSWithCache(ctx context.Context) (*clerk.JSONWebKeySet, error) {
+	cacheKey := "/v1/jwks"
```

Comment:
❓ Don't you need a mutex here?

Reply:
There's already a mutex when we read/write from the cache. I believe it should be enough?

Reply:
I think you're right. Since the JWKS is now cached it's not an issue if we perform multiple requests.